graylog2-web-interface/src/components/users/LdapComponent.jsx

```diff
@@ -0,0 +1,426 @@
+      </span>
+    ),
+    ADDITIONAL_GROUPS: (
+      'Choose the additional roles each Active Directory user will have by default, leave it empty if you want to map Active Directory groups to Graylog roles.'
+    ),
+  },
+
+  ldap: {
+    SYSTEM_USERNAME: (
+      <span>
+        The username for the initial connection to the LDAP server, e.g.{' '}
+        <code className="text-nowrap">uid=admin,ou=system</code>, this might be optional depending on your LDAP server.
+      </span>
+    ),
+    SYSTEM_PASSWORD: ('The password for the initial connection to the LDAP server.'),
+    SEARCH_BASE: (
+      <span>
+        The base tree to limit the LDAP search query to, e.g. <code
+        className="text-nowrap">cn=users,dc=example,dc=com</code>.
+      </span>
+    ),
+    SEARCH_PATTERN: (
+      <span>
+        For example <code className="text-nowrap">{'(&(objectClass=inetOrgPerson)(uid={0}))'}</code>.{' '}
+        The string <code>{'{0}'}</code> will be replaced by the entered username.
+      </span>
+    ),
+    DISPLAY_NAME: (
+      <span>
+        Which LDAP attribute to use for the full name of the user in Graylog, e.g. <code>cn</code>.<br/>
+        Try to load a test user using the form below, if you are unsure which attribute to use.
+      </span>
+    ),
+    GROUP_SEARCH_BASE: (
+      <span>
+        The base tree to limit the LDAP group search query to, e.g. <code>cn=users,dc=example,dc=com</code>.
+      </span>
+    ),
+    GROUP_PATTERN: (
+      <span>
+        The search pattern used to find groups in LDAP for mapping to Graylog roles, e.g.{' '}
+        <code>(objectClass=groupOfNames)</code> or{' '}
+        <code className="text-nowrap">(&amp;(objectClass=groupOfNames)(cn=graylog*))</code>.
+      </span>
+    ),
+    GROUP_ID: (
+      <span>Which LDAP attribute to use for the full name of the group, usually <code>cn</code>.</span>
+    ),
+    DEFAULT_GROUP: (
+      <span>
+        The default Graylog role determines whether a user created via LDAP can access the entire system, or has limited access.<br/>
+        You can assign additional permissions by{' '}
+        <LinkContainer to={Routes.SYSTEM.LDAP.GROUPS}><a>mapping LDAP groups to Graylog roles</a></LinkContainer>,{' '}
+        or you can assign additional Graylog roles to LDAP users below.
+      </span>
+    ),
+    ADDITIONAL_GROUPS: (
+      'Choose the additional roles each LDAP user will have by default, leave it empty if you want to map LDAP groups to Graylog roles.'
+    ),
+  },
+};
+
+const LdapComponent = React.createClass({
+  mixins: [Reflux.listenTo(LdapStore, '_onLdapSettingsChange', '_onLdapSettingsChange')],
+  getInitialState() {
+    return {
+      ldapSettings: undefined,
+      ldapUri: undefined,
+      roles: undefined,
+    };
+  },
+
+  componentDidMount() {
+    RolesStore.loadRoles().then(roles => {
+      this.setState({
+        roles: roles
+          .sort((r1, r2) => r1.name.localeCompare(r2.name))
```

Comment:
Could you please extract the logic processing the roles into a separate function? This way we would have to change less when we change the RolesStore to reflux.